client/windows/CreaturePurhaseCard.cpp

```diff
@@ -15,21 +15,43 @@
 #include "../../CCallback.h"
 #include "../CreatureCostBox.h"
 #include "QuickRecruitmentWindow.h"
+#include "../gui/CGuiHandler.h"
 
 void CreaturePurhaseCard::initButtons()
 {
 	initMaxButton();
 	initMinButton();
+	initCreatureSwitcherButton();
 }
 
 void CreaturePurhaseCard::initMaxButton()
 {
-	maxButton = std::make_shared<CButton>(Point(pos.x + 52, pos.y + 178), "iam014.def", CButton::tooltip(), std::bind(&CSlider::moveToMax,slider), SDLK_m);
+	maxButton = std::make_shared<CButton>(Point(pos.x + 52, pos.y + 178), "iam014.def", CButton::tooltip(), std::bind(&CSlider::moveToMax,slider));
 }
 
 void CreaturePurhaseCard::initMinButton()
 {
-	minButton = std::make_shared<CButton>(Point(pos.x, pos.y + 178), "iam015.def", CButton::tooltip(), std::bind(&CSlider::moveToMin,slider), SDLK_m);
+	minButton = std::make_shared<CButton>(Point(pos.x, pos.y + 178), "iam015.def", CButton::tooltip(), std::bind(&CSlider::moveToMin,slider));
+}
+
+void CreaturePurhaseCard::initCreatureSwitcherButton()
+{
+	creatureSwitcher = std::make_shared<CButton>(Point(pos.x + 20, pos.y-33), "iDv6432.def", CButton::tooltip(), [&](){ switchCreatureLevel(); });
+}
+
+void CreaturePurhaseCard::switchCreatureLevel()
+{
+	OBJECT_CONSTRUCTION_CAPTURING(ACTIVATE + DEACTIVATE + UPDATE + SHOWALL);
+	auto id = vstd::find(upgradesID, creatureOnTheCard->idNumber);
+	if(creatureOnTheCard->idNumber == upgradesID.back())
+		id = upgradesID.begin();
+	else
+		id++;
```

Comment:
What does `++` mean for creature id? [Upgrade id] =[Downgrade id] + 1 is true only for OH3 creatures by accident. It may not work with mods, (especially alternative upgrades)

Reply:
It's works for all mods, cove, forge, rampart from alternative upgrades also checked. So let look closer to it. First I copy into every "CreaturePurhaseCard" upgradesID that will store all id that the creature can be improved. Next, here I create iterator that have name "id". For it, ++ means only that I get next element from vector, in that case from "upgradesID". So nope, I don't get id+1. I get next id from upgradesID.

Reply:
I missed that `id` is an iterator.

Reply:
Will it more clear to do like that:
` auto index = vstd::find_pos(upgradesID, creatureOnTheCard->idNumber)`
`auto nextCreatureId = vstd::circular_at(upgradesID, ++index)`
?

Reply:
Did you mean vstd::circularAt? Yeah it's seems to be nicer solution.

Reply:
circularAt